src/Compilers/Core/Portable/Serialization/IObjectWritable.cs

```diff
@@ -9,5 +9,12 @@ namespace Roslyn.Utilities
     internal interface IObjectWritable
     {
         void WriteTo(ObjectWriter writer);
+
+        /// <summary>
+        /// Returns `true` when the same instance could be used more than once.
+        /// Instances that return "false" should not be tracked or the purpose 
+        /// of de-duplication while serializing/deserializing.
+        /// </summary>
+        bool IsReusable { get; }
```

Comment:
IsShareable might be better.

Reply:
`ShouldReuseInSerialization`  to cause no confusion.

---
In reply to: [181531123](https://github.com/dotnet/roslyn/pull/26156#discussion_r181531123) [](ancestors = 181531123)